### ISSUE TITLE
ENOMEM on uv_spawn on Windows with env set

### ISSUE THREAD

**Issue by SinisterRectus** [luvit/luv#386]
Code:
```lua
local uv = require('uv')

local command = 'cmd.exe'
local options = {
	env = {},
}

local handle, pid = uv.spawn(command, options, function(...)
	print('exit:', ...)
end)

print('spawn:', handle, pid)
```
Output:
```
spawn:  nil     ENOMEM: not enough memory
```

Removing the `env` field from the `options` table does not produce the issue. Tested with luv 1.27.0 via luvi 2.9.3 on Windows 10 Home 1903. Cannot reproduce on ubuntu 16.04 using an equivalent `command`. All 64-bit.

I don't know if this issue is in luv or in libuv or if it's fixed in a newer version. I ran into it while trying to build luvit on Windows and `test-process` and `test-process-exec` failed.

**Comment by squeek502** [comment]
Related: [this comments and the comments after it](https://github.com/luvit/luvit/pull/1063#issuecomment-507845192)

This test case should help a lot in getting to the bottom of it.

**Comment by squeek502** [comment]
This is a Libuv bug that was fixed in 1.31.0:

```
* win: fix uv_spawn() ENOMEM on empty env (Ben Noordhuis)
```

**Comment by SinisterRectus** [comment]
Good stuff.
